base/dict.jl

```diff
@@ -189,6 +189,16 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict, [element_type=Pair{K,V}])
```

Comment:
`K` and `V` are not defined in this signature. Rather use `eltype(d)`. Then you can give an example to show that one needs to pass a `Pair` type.